docs/roadmap.rst

```diff
@@ -70,22 +68,15 @@ environment and serve as a better foundation for testing for our Virtual Machine
 implementation.
 
 The client project will also build a solid foundation for continued internal research
-and development efforts. We plan to include new communication layer technologies
-in an exchangeable way - central will be support for Whisper (SHH) message communication
-and libp2p network topology, which will allow to contribute to various communication
-experiments and research on the way to a future sharded Ethereum network. Development
-in this direction is already in progress (libp2p).
+and development efforts. We've already incorporated support for libp2p as an alternate
+transport to RLPx/Devp2p that enables the in-browser light client use-case. In the future,
+we hope to implement a Clique PoA engine and test it on the Goerli testnet, and later build
+a working Ethereum 2.0 stateless client.
```

Comment:
Do you already have a very rough idea how much of the code base can eventually stay untouched on such a switch (I know, endless number of TBD, but anyhow 😄)? Rather 70% or rather 30%?

Reply:
@holgerd77 are you asking how much code change will be required to switch from PoW to a Clique PoA engine? if so, i think some of the work will be in implementing a PoA validator in ethereumjs-block and the rest will be in ethereumjs-client to implement the logic to actually sign blocks. It will mostly involve writing new code instead of touching existing code since the current code doesn't do much except download blocks :)

Reply:
Ah, thanks that's also interesting! 😄 No, I was rather asking on the 2.0 stateless client if you already got some intuition there if larger parts of the code should be reusable / generic enough for different client architectures.

Reply:
Ah! That is a great question :) 

I think the following components can be re-used without any changes: 

- `lib/net` (everything except the Protocol subclasses since 2.0 will require a new wire protocol)
- `lib/handler` (again, a new Handler subclass will need to be implemented for the new protocol)
- `lib/node` 

These components will probably have to modified or created:
- `lib/blockchain` (for the beacon chain, the blockchain class can probably be mostly re-used, but a new Block class will have to be defined that follow the new beacon chain Block format)
- `lib/sync` (the base classes can probably be re-used but new subclasses for fetchers and synchronizers will likely have to written for the beacon chain depending on the wire protocols)
- `lib/service` (again, the Service base class can probably be re-used but a new Ethereum2Service subclass will likely have to be written since we will need new protocols and synchronizers)

Reply:
Ah, thanks for this breakdown, very interesting! I actually always thought the 2.0 client would be intended for the shard chains and not the beacon chain. Was I mistaken?

Reply:
Actually, 2.0 light clients only need to download headers from the beacon chain, not the shard chains. I believe the only time they need to interact with a shard is if they need to lookup a specific state entry, in which case they will need to download a merkle branch and proof.

Reply:
Ah, if you sometime have very much leisure, maybe it's worth to write up your current state of understanding in a medium article. 😃 Not very much out there on stuff like this yet.

Reply:
Time is definitely limited for me, but I can definitely take a shot at it next time I get a break 😄

Reply:
Just some spontaneous idea, absolutely no need to pursue if you don't feel like it. 😄